File: erlang/easy/valid_strings.erl
-module(valid_strings).

-spec valid_strings(N :: integer()) -> [unicode:unicode_binary()].
valid_strings(N) ->
  valid_strings(0, 1, N).

valid_strings(Cur, _, N) when Cur =:= N ->
  1;
valid_strings(Cur, Pre, N) ->
  case Pre of
    $0 ->
      valid_strings(Cur + 1, 1, N);
    _ ->
      valid_strings(Cur + 1, 0, N) + valid_strings(Cur, 1, N)
  end.

File: go/middle/validStrings.go
package main

func validStrings(n int) []string {
	rs := make([]string, 0)
	var recur func(cur string)
	recur = func(cur string) {
		if len(cur) == n {
			rs = append(rs, cur)
			return
		}
		if len(cur) == 0 || cur[len(cur)-1] == '1' {
			recur(cur + "0")
			recur(cur + "1")
		} else {
			recur(cur + "1")
		}
	}
	return rs
}
